### DISCUSSION TITLE
Milestone 2.0.0 Release Scope

### DISCUSSION THREAD

**Discussion by andregasser** [andregasser/bigbone#29] [General]
Let's have the scope discussion for the 2.0.0 milestone release in here. I have already started to update [API.md](https://github.com/andregasser/mastodon4j/blob/master/API.md) to get an overview which Mastodon API endpoints are covered by the library and which not (fully, partial, not at all).

**Comment by bocops** [comment]
From looking at the issues that have been created so far, also related discussions, it seems as if there's a number of distinct areas of improvement at the moment. In no specific order:

1. Complete coverage of already implemented endpoints and entities (issues 14 to #24)
2. Implement new/missing endpoints (#13)
3. Rename project and package structure (#31)
4. Hide implementation details like OkHttp and Gson (#5)
5. Update dependencies (#3)
6. General cleanup (my patch in #5; things like #11)
7. Bugfixes (#25)

We could do all of that at once, which might take a while, or prioritize to get something out there quickly as 2.0.0 and then do the rest in subsequent major releases. If we're listed on joinmastodon.org with a first "proper" release, we might get some more help from others.

I think "Renaming" should definitely happen now, no matter what else we do for 2.0.0.

**Reply by andregasser** [reply]
That's a valid point. According to the current API.md, a lot of endpoints are not implemented or not fully implemented at the moment. Let's reduce the scope of 2.0.0. I think 1 and 2 in your list could be done in a later release (it is just adding new features). 3 is a must have. 4 is also Ok for me in a 2.0.0 release (is sort of cleanup). 5 would be nice as well, as the current 2.x version is unsupported. 6 is also ok for the 2.0.0 release and issue #25 mentioned in point 7 is a must I think as there were some other users that have reported issues with that on the sys1yagi/mastodon4j repository.

If this is ok for you, we could adjust the 2.0.0 milestone scope. 

I have added a Project "2.0.0 Release" to this repo to track implementation progress.

What's your opinion regarding the artifact repository? Shall we stay with JitPack or maybe switch to MavenCentral for the first release?

**Reply by bocops** [reply]
So 3 and 7 as the bare minimum, and as much progress on 4-6 as sensible, sounds good to me. :)

Regarding artifact repository, I have no strong personal opinion either way - but MavenCentral is already specified as a default repository in new Android Studio projects, which might make adoption ever so slightly easier for the Android app folks.

**Comment by mkarg** [comment]
Is ther a publicly visible plan somewhere how far away 2.0.0 is?

**Reply by PattaFeuFeu** [reply]
Not currently, no. If possible, I’d like to have a 2.0.0 release this quarter but we have yet to get an up-to-date overview on what is currently missing and still needs to be done.

That would entail:
* Finalising our public API including decisions on whether we want to use `data class`es with overlapping attributes for e.g. `Account`, `MutedAccount`, `CredentialsAccount` VS switch to regular classes and inheritance
* Setting up the release target anew: With the owner change and a domain about to expire, we’ll have to set up MavenCentral or another artifact hoster first
* Ensuring we have all endpoints covered and all attributes in all entities supported. That’s of course an ongoing task but I’d like us to have everything supported at the point of finally releasing

among others.
